contrib/build_executable.jl

```diff
@@ -0,0 +1,248 @@
+    run(`rm -rf $(tmpdir) $(sys.buildfile).o $(sys.buildfile0).o $(sys.buildfile0).ji`)
+    println()
+
+    if targetdir != nothing
+        # Move created files to target directory
+        run(`mv $(exe_release.buildfile) $(exe_debug.buildfile) $(sys.buildfile).$(Sys.dlext) $(sys.buildfile).ji $(targetdir)`)
+
+        # Copy needed shared libraries to the target directory
+        tmp = ".*\.$(Sys.dlext).*"
+        shlibs = filter(Regex(tmp),readdir(sys.buildpath))
+        for shlib in shlibs
+            run(`cp $(joinpath(sys.buildpath, shlib)) $(targetdir)`)
+        end
+
+        # Fix rpath in executable and shared libraries
+        shlibs = filter(Regex(tmp),readdir(targetdir))
+        push!(shlibs, exe_release.filename)
+        push!(shlibs, exe_debug.filename)
+
+        for shlib in shlibs
+            rpath = readall(`$(patchelf) --print-rpath $(joinpath(targetdir, shlib))`)[1:end-1]
+            if Base.samefile(rpath, sys.buildpath)
+                run(`$(patchelf) --set-rpath $(targetdir) $(joinpath(targetdir, shlib))`)
+            end
+        end
+    end
+
+    for exe in [exe_release, exe_debug]
+        println("$(exe.targetfile) successfully created.")
+    end
+    return 0
+end
+
+function find_patchelf()
+    patchelf = nothing
+
+    for x in [joinpath(JULIA_HOME, "patchelf"), "patchelf"]
+        patchelf = x
+        works = true
+        try
+            x=readall(`$(patchelf) --version`)
+        catch
+            works = false
+        end
+        works && break
+    end
+
+    patchelf
+end
+
+function get_includes()
+    ret = []
+
+    # binary install
+    incpath = abspath(joinpath(JULIA_HOME, "..", "include", "julia"))
+    push!(ret, "-I$(incpath)")
+
+    # Git checkout
+    julia_root = abspath(joinpath(JULIA_HOME, "..", ".."))
+    push!(ret, "-I$(julia_root)src")
+    push!(ret, "-I$(julia_root)src/support")
+    push!(ret, "-I$(julia_root)usr/include")
+
+    ret
+end
+
+function emit_cmain(cfile, exename, relocation)
+    if relocation
+        sysji = joinpath("lib"*exename)
+    else
+        sysji = joinpath("..", "lib", "lib"*exename)
+    end
+
+    f = open(cfile, "w")
+    write( f, """
+        #include <julia.h>
+        #include <stdio.h>
+
+        int main(int argc, char *argv[])
+        {
+            char sysji[] = \"$(sysji).ji\";
+            char mainfunc[] = \"main()\";
+            jl_init_with_image(NULL, sysji);
+            jl_eval_string(mainfunc);
+
+            int ret = 0;
+            if (jl_exception_occurred())
+            {
+                jl_show(jl_stderr_obj(), jl_exception_occurred());
+                JL_PRINTF(jl_stderr_stream(), \"\\n\");
+                ret = 1;
+            }
+
+            jl_atexit_hook();
+            return ret;
+        }
+        """
+    )
+    close(f)
+end
+
+function emit_userimgjl(userimgjl, script_file)
+    f = open(userimgjl, "w")
+    write( f, """
+        include(\"$(script_file)\")
```

Comment:
This line should look like: `include(\"$(escape_string(script_file))\")` in order for the path to the script file is correct on windows.